dxp/taglib_aui_input.md

```diff
@@ -0,0 +1,1409 @@
+
+It's always recommended to pass a value to the `label` attribute. Use `label=""` if you want to output an input without a `<label>` element.
+
+The value of the `label` attribute gets passed to the tag `<liferay-ui:message key="<%= label %>" localizeKey="<%= localizeLabel %>" />` which takes language keys and translates them for us.
+
+## Text
+
+> This section also applies to `input` types `color`, `password`, `url`, `tel`, `email`, `search`, `number`, and `range`.
+
+Any attribute you pass to `<aui:input type="text" />` will be output. There are a handful of attributes that the tag processes before being sent to the browser. They are listed in the table below along with other Liferay specific attributes:
+
+| Attribute                                              | Description                                                                                                                                                                                                                                                                            |
+| ------------------------------------------------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| <code id="text-autoFocus">autoFocus</code>             | A `boolean` value that focuses the input on page load (e.g., `autoFocus="<%= true %>"`). This might not work with Single Page Applications.                                                                                                                                            |
+| <code id="text-bean">bean</code>                       | TODO: A bean to associate with the input component.                                                                                                                                                                                                                                    |
+| <code id="text-cssClass">cssClass</code>               | Adds classes to `<input class="form-control" />`                                                                                                                                                                                                                                       |
+| <code id="text-data">data</code>                       | A convenience attribute for pumping out data attributes from `key, value` pairs in a Java `HashMap`. For example, the `HashMap<"class-name", foo>` would render as the attribute `data-class-name = foo;`. You can also use the standard HTML data attribute `data-class-name="foo"`.  |
```

Comment:
Probably: 

`HashMap<"class-name", foo>` -> `HashMap.Entry<"class-name", foo>`  

would be more legible.